microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/config/java/MicroProfileConfigASTValidator.java

```diff
@@ -224,36 +225,70 @@ private boolean isPropertyIgnored(String propertyName) {
 		return false;
 	}
 
+	private static final Pattern ARRAY_SPLITTER = Pattern.compile("(?<!\\\\),");
+
 	private static boolean isAssignable(ITypeBinding fieldBinding, IJavaProject javaProject, String defValue) {
 		String fqn = Signature.getTypeErasure(fieldBinding.getQualifiedName());
+		// handle list-like types.
+		// MicroProfile config supports arrays, `java.util.List`, and `java.util.Set` by
+		// default. See:
+		// https://download.eclipse.org/microprofile/microprofile-config-2.0/microprofile-config-spec-2.0.html#_array_converters
+		if (fieldBinding.isArray()) {

```

Comment:
When an `ITypeBinding` is an array, does it work to call `getElementType()` to return the element type of the array ? At least to avoid the string manipulation here.